packages/mui-base/src/useSelect/useSelect.ts

```diff
@@ -27,6 +27,44 @@ import { selectReducer } from './selectReducer';
 import { combineHooksSlotProps } from '../utils/combineHooksSlotProps';
 import { MuiCancellableEvent } from '../utils/MuiCancellableEvent';
 
+// visually hidden style based on https://webaim.org/techniques/css/invisiblecontent/
+const visuallyHiddenStyle: React.CSSProperties = {
+  clip: 'rect(1px, 1px, 1px, 1px)',
+  clipPath: 'inset(50%)',
+  height: '1px',
+  width: '1px',
+  margin: '-1px',
+  overflow: 'hidden',
+  padding: 0,
+  position: 'absolute',
+  left: '50%',
+  bottom: 0, // to display the native browser validation error at the bottom of the Select.
+};
+
+const noop = () => {};
+
+function defaultFormValueProvider<OptionValue>(
+  selectedOption: SelectOption<OptionValue> | SelectOption<OptionValue>[] | null,
+) {
+  if (Array.isArray(selectedOption)) {
+    if (selectedOption.length === 0) {
+      return '';
+    }
+
```

Comment:
I deleted this because it breaks the new tests I added:

```js
  if (
      selectedOption.every(
        (o) =>
          typeof o.value === 'string' ||
          typeof o.value === 'number' ||
          typeof o.value === 'boolean',
      )
    ) {
      return selectedOption.map((o) => String(o.value));
    }
```

@michaldudak Do you know why it existed?

Reply:
Numbers and booleans are easily converted to strings with `String()`, so if all selected options are of these types (or strings), we can simply convert them and fall back to JSON when options are objects.

Reply:
I believe that it should be removed, otherwise, the return type will be `string | string[]` which defeats the purpose of serialized value.

Reply:
Yeah, OK, right. I don't remember the rationale for allowing string[], but what you're saying makes sense.